api/tests/identity-access-management/unit/domain/usecases/create-access-token-from-refresh-token.usecase.test.js

```diff
@@ -109,4 +110,82 @@ describe('Unit | Identity Access Management | Domain | UseCases | create-access-
       expect(error.code).to.equal('INVALID_REFRESH_TOKEN');
     });
   });
+
+  context('when the refresh token audience is not the same', function () {
+    context('when access token confinement feature is enabled', function () {
+      it('throws an unauthorized error', async function () {
+        // given
+        sinon.stub(config.featureToggles, 'isUserTokenAudConfinementEnabled').value(true);
+
+        const userId = '123';
+        const accessToken = 'valid access token';
+        const expirationDelaySeconds = 1;
+        const scope = 'mon-pix';
+        const source = 'pix';
+        const audience = 'https://app.pix.fr';
+        const badAudience = 'https://orga.pix.fr';
+
+        const refreshToken = RefreshToken.generate({ userId, scope, source, audience });
+
+        const refreshTokenRepository = { findByToken: sinon.stub() };
+        refreshTokenRepository.findByToken.withArgs({ token: refreshToken.value }).resolves(refreshToken);
+
+        const tokenService = { createAccessTokenFromUser: sinon.stub() };
+        tokenService.createAccessTokenFromUser
+          .withArgs({ userId, source, audience })
+          .resolves({ accessToken, expirationDelaySeconds });
+
+        // when
+        const error = await catchErr(createAccessTokenFromRefreshToken)({
+          refreshToken: refreshToken.value,
+          scope,
+          refreshTokenRepository,
+          tokenService,
+          audience: badAudience,
+        });
+
+        // then
+        expect(error).to.instanceOf(UnauthorizedError);
```

Comment:
Suggestion: faire un seul expect

```suggestion
expect(error).to.deepEqualInstance(new UnauthorizedError({message: 'Refresh token is invalid', code: 'INVALID_REFRESH_TOKEN'}));```